apps/services/auth/api/src/app/v1/delegations/delegations.controller.spec.ts

```diff
@@ -204,10 +228,18 @@ describe('DelegationsController', () => {
             nationalRegistryApiSpy = jest
               .spyOn(nationalRegistryApi, 'getIndividual')
               .mockImplementation((id) => {
+                if (deceasedNationalIds.includes(id)) {
+                  return Promise.resolve(null)
+                }
+
                 const user = nationalRegistryUsers.find(
-                  (u) => u.nationalId === id,
+                  (u) =>
+                    u?.nationalId === id &&
+                    // Make sure we don't return a user that has been marked as deceased
+                    !deceasedNationalIds.includes(u?.nationalId),
                 )
-                return user ? Promise.resolve(user) : Promise.reject()
+
+                return user ? Promise.resolve(user) : Promise.reject(null)
```

Comment:
Should resolve with null. This whole function would probably be simpler as an `async` function.

Reply:
good point. Old code as well